### PR TITLE
Multi threading examples (tasks, queues, semaphores, mutexes)

### PR DESCRIPTION
**Pull request by PilnyTomas**
Moved the original ESP32/FreeRTOS example to a new pseudo library (no actual source code) called MultiThreading.
The original example was renamed to BasicMultiThreading demonstrating the most basic use of tasks.
New examples will be added demonstrating the use of queues, semaphores, and mutexes.
As part of this PR I will also adhere to the new standards - add readme files, headers, etc stated here https://github.com/espressif/arduino-esp32/issues/7659
Related https://github.com/espressif/arduino-esp32/issues/228

### CODE REVIEW

**Comment by mrengineer7777** [comment]
Queue.ino example may be helpful in a future project.  Thanks!

File: libraries/MultiThreading/examples/BasicMultiThreading/BasicMultiThreading.ino [hunk 1]
```diff
@@ -0,0 +1,105 @@
+// Please read file README.md in the folder containing this example.
```
Comment (pedrominatel) [inline libraries/MultiThreading/examples/BasicMultiThreading/BasicMultiThreading.ino:1]: Please add the header.

```
/* Basic Multi Threading Arduino Example
   This example code is in the Public Domain (or CC0 licensed, at your option.)
   Unless required by applicable law or agreed to in writing, this
   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied.
*/
```

Same for the others.

File: libraries/MultiThreading/examples/Semaphore/Semaphore.ino [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Please read file README.md in the folder containing this example.
```
Comment (pedrominatel) [inline libraries/MultiThreading/examples/Semaphore/Semaphore.ino:1]: This example is not working on my ESP32 board.

`Backtrace: 0x40087e51:0x3ffb9150 0x40084557:0x3ffb9230 0x400d2343:0x3ffb9270 0x400d157d:0x3ffb9290 0x400d1903:0x3ffb92b0 0x400d1915:0x3ffb92d0 0x400d12b3:0x3ffb92f0`

File: libraries/MultiThreading/examples/BasicMultiThreading/BasicMultiThreading.ino [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/* Basic Multi Threading Arduino Example
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+// Please read file README.md in the folder containing this example.
+
```
Comment (P-R-O-C-H-Y) [inline libraries/MultiThreading/examples/BasicMultiThreading/BasicMultiThreading.ino:8]: This example is not working for me on ESP32-S3 devkitC. Can you please test that? 
Analog pin was changed of course.

```
Guru Meditation Error: Core  1 panic'ed (Unhandled debug exception). 
Debug exception reason: Stack canary watchpoint triggered (Analog Read) 

Backtrace: 0x4037d1bc:0x3fced4e0 0x4037bdfd:0x3fced520 0x4037a71c:0x3fced550 0x4037a712:0xa5a5a5a5 |<-CORRUPTED
```

File: libraries/MultiThreading/examples/BasicMultiThreading/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+Before creating a new issue, be sure to try Troubleshooting and check if the same issue was already created by someone else.
+
+## Resources
+
+* Official ESP32 Forum: [Link](https://esp32.com)
+* Arduino-ESP32 Official Repository: [espressif/arduino-esp32](https://github.com/espressif/arduino-esp32)
+* ESP32 Datasheet: [Link to datasheet](https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf)
+* ESP32-S2 Datasheet: [Link to datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-s2_datasheet_en.pdf)
+* ESP32-C3 Datasheet: [Link to datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-c3_datasheet_en.pdf)
```
Comment (P-R-O-C-H-Y) [inline libraries/MultiThreading/examples/BasicMultiThreading/README.md:87]: Please add ESP32-S3 datasheet link to all README files. PR https://github.com/espressif/arduino-esp32/pull/7738.

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@PilnyTomas Please take a look on my comments.

**Comment by P-R-O-C-H-Y** [comment]
One more comment, `Semaphore.ino` is missing Readme. As you did nice readme with theory and explanation, it would be great to have it for semaphore too even it's simpler.

**Comment by PilnyTomas** [comment]
> One more comment @PilnyTomas, `Semaphore.ino` is missing Readme. As you did nice readme with theory and explanation, it would be great to have it for semaphore too even it's simpler.

I just forgot to add it :D

**Review by pedrominatel — APPROVED** [review]
LGTM

**Review by P-R-O-C-H-Y — APPROVED** [review]
All examples are now working :)
Just please @PilnyTomas add the ESP32-S3 data sheets link into all readme files. Thanks

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM!

**Comment by me-no-dev** [comment]
I will suggest to move all new examples into `libraries/ESP32/examples/FreeRTOS/` folder. They do not need to be in their own library

**Review by me-no-dev — CHANGES_REQUESTED** [review]
Change location of examples to `libraries/ESP32/examples/FreeRTOS/`

**Review by me-no-dev — APPROVED** [review]

**Pull request merged**